### PR TITLE
statistics: make sure PQ can analyze all indexes with stats version 1

### PR DESCRIPTION
**Pull request by Rustin170506**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/57257

Problem Summary:

### What changed and how does it work?

After switching to the new PQ, we only have one chance to analyze newly added indexes. For example, if two indexes are created in a very short period and the analyze version 1 is used, we need to analyze these indexes separately.

However, with the current implementation, we only analyze one index at a time, which means the second index will be missed.

In this PR, I’ve checked the analyze version to ensure proper handling. The lower versions don’t have this issue because we can analyze the second index after rebuilding the entire priority queue.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by elsa0520 — APPROVED** [review]
LGTM

File: pkg/statistics/handle/autoanalyze/priorityqueue/non_partitioned_table_analysis_job.go [hunk 1]
```diff
@@ -212,6 +212,16 @@ func (j *NonPartitionedTableAnalysisJob) analyzeIndexes(
 	if len(j.Indexes) == 0 {
 		return true
 	}
+	autoAnalyzeVersion := sctx.GetSessionVars().AnalyzeVersion
+	if autoAnalyzeVersion == 1 {
```
Comment (elsa0520) [inline pkg/statistics/handle/autoanalyze/priorityqueue/non_partitioned_table_analysis_job.go:216]: Please add some comments to explain why we need to Special treatment the auto analyze version 1 ?

Comment (Rustin170506) [reply pkg/statistics/handle/autoanalyze/priorityqueue/non_partitioned_table_analysis_job.go]: Updated. Thank you!

**Review by qw4990 — APPROVED** [review]

**Comment by Rustin170506** [comment]
@ti-chi-bot Wake up

**Pull request merged**